test/core.js

```diff
@@ -105,104 +69,64 @@ test( "attribute selectors with naked '#'", function() {
 	} );
 } );
 
-test( "selector state", function() {
-	expect( 18 );
-
-	var test;
-
-	test = jQuery( undefined );
-	equal( test.selector, "", "Empty jQuery Selector" );
-	equal( test.context, undefined, "Empty jQuery Context" );
-
-	test = jQuery( document );
-	equal( test.selector, "", "Document Selector" );
-	equal( test.context, document, "Document Context" );
-
-	test = jQuery( document.body );
-	equal( test.selector, "", "Body Selector" );
-	equal( test.context, document.body, "Body Context" );
-
-	test = jQuery( "#qunit-fixture" );
-	equal( test.selector, "#qunit-fixture", "#qunit-fixture Selector" );
-	equal( test.context, document, "#qunit-fixture Context" );
-
-	test = jQuery( "#notfoundnono" );
-	equal( test.selector, "#notfoundnono", "#notfoundnono Selector" );
-	equal( test.context, document, "#notfoundnono Context" );
-
-	test = jQuery( "#qunit-fixture", document );
-	equal( test.selector, "#qunit-fixture", "#qunit-fixture Selector" );
-	equal( test.context, document, "#qunit-fixture Context" );
+test( "XSS injection (leading hash)", function() {
+	expect( 1 );
 
-	test = jQuery( "#qunit-fixture", document.body );
-	equal( test.selector, "#qunit-fixture", "#qunit-fixture Selector" );
-	equal( test.context, document.body, "#qunit-fixture Context" );
+	var threw = false;
 
-	// Test cloning
-	test = jQuery( test );
-	equal( test.selector, "#qunit-fixture", "#qunit-fixture Selector" );
-	equal( test.context, document.body, "#qunit-fixture Context" );
+	try {
+		jQuery( "#yeah<p>RIGHT</p>" );
+	} catch ( e ) {
+		threw = true;
+	}
 
-	test = jQuery( document.body ).find( "#qunit-fixture" );
-	equal( test.selector, "#qunit-fixture", "#qunit-fixture find Selector" );
-	equal( test.context, document.body, "#qunit-fixture find Context" );
+	equal( threw, true, "Throw on leading-hash HTML (treated as selector)" );
 } );
 
-test( "XSS injection", function() {
-	expect( 10 );
+test( "XSS injection (XSS via script tag)", function() {
```

Comment:
I thought about it but I thought it might be worth to keep them as long as we patch `jQuery.fn.init` in any way, just in case. No strong opinions here, though.